Cubical/Categories/LocallySmall/NaturalTransformation/Fibered.agda

```diff
@@ -0,0 +1,223 @@
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.HLevels.More
+open import Cubical.Foundations.Isomorphism hiding (isIso)
+
+import Cubical.Data.Equality as Eq
+open import Cubical.Data.Sigma
+open import Cubical.Data.Sigma.More
+
+open import Cubical.Reflection.RecordEquiv.More
+
+import Cubical.Categories.Category as Small
+import Cubical.Categories.Functor as SmallFunctor
+open import Cubical.Categories.LocallySmall.Category.Base
+open import Cubical.Categories.LocallySmall.Category.Small
+open import Cubical.Categories.LocallySmall.Functor.Base
+open import Cubical.Categories.LocallySmall.Variables
+
+open import Cubical.Categories.LocallySmall.Displayed.Category.Base
+open import Cubical.Categories.LocallySmall.Displayed.Category.Notation
+open import Cubical.Categories.LocallySmall.Displayed.Category.Small
+open import Cubical.Categories.LocallySmall.Displayed.Category.Properties
+open import Cubical.Categories.LocallySmall.Displayed.Constructions.Reindex.Base
+open import Cubical.Categories.LocallySmall.Displayed.Constructions.Reindex.Properties
+
+open Category
+open Categoryᴰ
+open SmallCategory
+
+module FiberedFunctorDefs
+  (C : SmallCategory ℓC ℓC')
+  {D : Category Dob DHom-ℓ}
+  {Dobᴰ-ℓ Dobᴰ DHom-ℓᴰ}
+  (Dᴰ : SmallFibersCategoryᴰ D Dobᴰ-ℓ Dobᴰ DHom-ℓᴰ)
+  where
+  private
+    module C = SmallCategory C
+    module D = Category D
+    module Dᴰ = CategoryᴰNotation Dᴰ
+
+  FiberedFunctor : (d : Dob) → Typeω
+  FiberedFunctor d = Functor C.cat Dᴰ.v[ d ]
+
+  module _ (D-⋆ : ∀ {x} → D.id D.⋆ D.id Eq.≡ D.id {x}) (d : Dob) where
+    FiberedFunctorEq : Typeω
+    FiberedFunctorEq = Functor C.cat (fibEq Dᴰ D-⋆ d)
+
+    FiberedFunctor→FiberedFunctorEq :
+      FiberedFunctor d → FiberedFunctorEq
+    FiberedFunctor→FiberedFunctorEq = fib→fibEq Dᴰ D-⋆ d ∘F_
+
+    FiberedFunctorEq→FiberedFunctor :
+      FiberedFunctorEq → FiberedFunctor d
+    FiberedFunctorEq→FiberedFunctor = fibEq→fib Dᴰ D-⋆ d ∘F_
+
+  module FiberedFunctorNotation {d : Dob} (F : FiberedFunctor d)
+    where
+
+    open FunctorNotation F public
+
+module FibNatTransDefs
+  (C : SmallCategory ℓC ℓC')
+  {D : Category Dob DHom-ℓ}
+  {Dobᴰ-ℓ Dobᴰ DHom-ℓᴰ}
+  (Dᴰ : SmallFibersCategoryᴰ D Dobᴰ-ℓ Dobᴰ DHom-ℓᴰ)
+  where
+  private
+    module C = SmallCategory C
+    module D = CategoryNotation D
+    module Dᴰ = CategoryᴰNotation Dᴰ
+  open FiberedFunctorDefs C Dᴰ public
+  module _
+    {d d' : Dob} (g : D.Hom[ d , d' ])
+    (F : FiberedFunctor d)
+    (G : FiberedFunctor d')
+    where
+    private
+      module F = FiberedFunctorNotation F
+      module G = FiberedFunctorNotation G
+
+    N-homTy :
+      (N-ob : ∀ x → Dᴰ.Hom[ g ][ F.F-ob (liftω x) , G.F-ob (liftω x) ])
+      → ∀ {x y} → (f  : C.Hom[ liftω x , liftω y ]) → Type _
+    N-homTy N-ob {x} {y} f =
+      (F.F-hom f Dᴰ.⋆ⱽᴰ N-ob y) ≡
```

Comment:
Based on playing with this in Agda the use of ⋆ⱽᴰ seems to be the source of all your issues. If you use
```
      (F.F-hom f Dᴰ.⋆ᴰ N-ob y) Dᴰ.≡[ D.⋆IdL g ∙ (sym $ D.⋆IdR g) ] (N-ob x Dᴰ.⋆ᴰ G.F-hom f)
```
I think you will have the exact same universe levels but without the transports.
OTOH you can also use
```
      (F.F-hom f Dᴰ.⋆ᴰ N-ob y) Dᴰ.∫≡ (N-ob x Dᴰ.⋆ᴰ G.F-hom f)
```
which will be more ergonomic but live at a higher universe level.